containerd-shim-spin/src/engine.rs

```diff
@@ -156,14 +157,18 @@ impl SpinEngine {
         let cache = initialize_cache().await?;
         let app_source = Source::from_ctx(ctx, &cache).await?;
         let mut locked_app = app_source.to_locked_app(&cache).await?;
-        let components_to_execute = env::var(constants::SPIN_COMPONENTS_TO_RETAIN_ENV)
-            .ok()
-            .map(|s| s.split(',').map(|s| s.to_string()).collect::<Vec<String>>());
-        if let Some(components) = components_to_execute {
-            if let Err(e) = crate::retain::retain_components(&mut locked_app, &components) {
-                println!("Error with selective deployment: {:?}", e);
-                return Err(e);
-            }
+        if let Ok(components_env) = env::var(constants::SPIN_COMPONENTS_TO_RETAIN_ENV) {
+            let components = components_env.split(',').collect::<Vec<&str>>();
```

Comment:
question: do you need to filter empty values?

Reply:
Good check. Added